extensions/workloads/template.go

```diff
@@ -118,3 +118,19 @@ func NewJobTemplate(jobName string, namespace string) *batchv1.Job {
 	}
 
 }
+
+// NewPodTemplateWithSecretEnvironmentVariable is a constructor that returns pod template spec with envFrom option for workload creations
+func NewPodTemplateWithSecretEnvironmentVariable(containerName string, containerImage string, secretName string) corev1.PodTemplateSpec {
```

Comment:
Do you think this function going to be widely used? I think we already have the generic templates to do this, and that's essentially what this function does. It seems more like a private helper.

This might mean we could get different `NewPodTemplateX` for different configurations. What would be the pros/cons?

Reply:
This template is already available in our codebase here: https://github.com/rancher/rancher/blob/b421ea5e0bfeaf899bb28141db85412da68ae49e/tests/v2/validation/upgrade/workload.go#L151 
I need this for my new tests that I plan to implement for a new feature called project scoped secrets where, I need to create a deployment using secret as environment variable. 
So instead of duplicating it, It'd be beneficial to have this as a extension instead since it is already being used in one of the upgrade tests and I also need this for my new tests. By having it as an extension, we can avoid duplication.

Reply:
The feature "Project scoped secrets" that was going to use this helper function has been dropped. So, we no longer need this helper function and has been removed.